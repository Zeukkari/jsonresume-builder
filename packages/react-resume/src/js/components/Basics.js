import React from 'react'

import { Box, Button, Heading, Paragraph, Text, Image } from 'grommet'

import { TopLayout } from './Layout'

import { ResumeSection } from './common'

export const Picture = ({ picture }) => (
  <Box align="center" justify="center">
    <Image src={picture} size="large" />
  </Box>
)
Picture.defaultProps = {
  picture: null,
}

const Profiles = ({ profiles }) => {
  const profileButtons = profiles.map(profile => (
    <Button label={profile.network} key={profile.network} />
  ))

  return <Box direction="row">{profileButtons}</Box>
}

export const Contact = ({ email, phone, location, website, profiles }) => {
  const locationString = `${location.address} ${location.postalCode} ${
    location.city
  } ${location.countryCode}`

  return (
    <Box
      flex
      align="center"
      justify="between"
      direction="row"
      pad="none"
      margin="none"
      border={{
        side: 'all',
        color: 'red',
        size: 'small',
        style: 'groove',
      }}
    >
      <Text> email: {email} </Text>

      <Text> website: {website} </Text>

      <Text> Location: {locationString} </Text>

      <Text> Phone: {phone} </Text>

      <Profiles profiles={profiles} />
    </Box>
  )
}
Contact.defaultProps = {
  email: 'N/A',
  phone: 'N/A',
  website: 'N/A',

  location: {
    address: 'No address',
    postalCode: '0',
    city: 'N/A',
    countryCode: 'FI',
    region: 'N/A',
  },
  profiles: [
    {
      network: 'Twitter',
      username: 'john',
      url: 'about:blank',
    },
  ],
}

export const Summary = ({ name, label, summary }) => {
  return (
    <Box
      align="center"
      border={{
        side: 'all',
        color: 'red',
        size: 'xsmall',
        style: 'dashed',
      }}
    >
      <ResumeSection>
        <Heading level={1} size="small" margin="xsmall" textAlign="center">
          {name}
        </Heading>
        <Heading level={2} size="small" margin="xsmall" textAlign="center">
          {label}
        </Heading>
        <Box>
          <Paragraph size="medium">{summary}</Paragraph>
        </Box>
      </ResumeSection>
    </Box>
  )
}
Summary.defaultProps = {
  name: 'N/A',
  label: 'N/A',
  summary: 'N/A',
}

export const Basics = ({
  name,
  label,
  summary,
  email,
  phone,
  location,
  website,
  profiles,
  picture,
}) => (
  <TopLayout>
    {picture && (
      <Box align="center" justify="center">
        <Picture picture={picture} />
      </Box>
    )}

    <Summary name={name} label={label} summary={summary} />

    <Contact
      email={email}
      phone={phone}
      website={website}
      location={location}
      profiles={profiles}
    />
  </TopLayout>
)
