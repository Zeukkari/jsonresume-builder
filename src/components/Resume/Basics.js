import React from 'react';

import {
  Box,
  Heading,
  Paragraph,
} from 'grommet';

const Basics = ({ basics }) => {
  const name = basics.name || 'N/A';
  const label = basics.label || 'N/A';
  const summary = basics.summary || 'N/A';
  const email = basics.email || 'N/A';
  const website = basics.website || 'N/A';
  // const picture = basics.picture || 'N/A';
  // const profiles = basics.profiles || [];


  return (
    <Box>
      <Heading> Basics </Heading>
      <Paragraph>name: {name}</Paragraph>
      <Paragraph>label: {label}</Paragraph>
      <Paragraph>summary: {summary}</Paragraph>
      <Paragraph>email: {email} </Paragraph>
      <Paragraph>website: {website}</Paragraph>
    </Box>
  );
};

export default Basics;