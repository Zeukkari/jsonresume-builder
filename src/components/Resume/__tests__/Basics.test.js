import React from 'react';
import renderer from 'react-test-renderer';

import Basics from '../Basics';

import defaultResume from '../../../util/defaultResume';


test('renders basics section', () => {
  const data = defaultResume.basics;
  const component = renderer.create(
    <Basics basics={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});