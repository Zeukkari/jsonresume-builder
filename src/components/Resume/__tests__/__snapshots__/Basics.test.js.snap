// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basics section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 dCmVAK"
>
  <h1
    className="StyledHeading-sc-1rdh4aw-0 jVpYfH"
  >
     Basics 
  </h1>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    name: 
    N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    label: 
    N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    Summary: 
    N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    summary: 
    N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    email: 
    N/A
     
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    website: 
    N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    Location: 
    No address 0 N/A FI N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    Picture: 
    N/A
  </p>
  <p
    className="StyledParagraph-tbetod-0 kxFtnQ"
  >
    Phone: 
    N/A
  </p>
  <table
    className="StyledDataTable-xrlyjm-0 kTrXKK"
  >
    <thead
      className="StyledDataTable__StyledDataTableHeader-xrlyjm-3 bAIczD StyledTable__StyledTableHeader-sc-1m3u5g-4 dHaGyd"
    >
      <tr
        className="StyledDataTable__StyledDataTableRow-xrlyjm-1 ezpFws StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Network
            </span>
          </div>
        </th>
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Username
            </span>
          </div>
        </th>
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              URL
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      className="StyledDataTable__StyledDataTableBody-xrlyjm-2 hbZQKY"
    >
      <tr
        className="StyledDataTable__StyledDataTableRow-xrlyjm-1 ezpFws"
      >
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="row"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <span
              className="StyledText-sc-1sadyjn-0 dndifk"
            >
              Twitter
            </span>
          </div>
        </th>
        <td
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              john
            </span>
          </div>
        </td>
        <td
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              about:blank
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
